.github/workflows/tarball-check.yml

```diff
@@ -0,0 +1,52 @@
+name: Check Tarball
```

Comment:
let's start naming these actions consistently. we have "SBT tests" (noun) and "Check Docs" (verb). Lets make them all nouns 
- SBT Tests
- Doc Tests
- Binary Generation Tests